### PR TITLE
Add Tangent Space Alignment

### PR DESCRIPTION
**Pull request by qbarthelemy**
Deep refacto of code shared in #319.

### CODE REVIEW

File: pyriemann/transfer/_estimators.py [hunk 1]
```diff
@@ -592,43 +601,474 @@ def fit_transform(self, X, y_enc, sample_weight=None):
...
+        return X_new
+
+
+###############################################################################
+
+
+class TlTsCenter(BaseEstimator, TransformerMixin):
+    """Center vectors in tangent space for transfer learning.
+
+    Center the vectors from each domain to zero on tangent space.
```
Comment (plcrodrigues) [inline pyriemann/transfer/_estimators.py:628]: To the origin of tangent space

File: pyriemann/transfer/_estimators.py [hunk 2]
```diff
@@ -592,43 +601,474 @@ def fit_transform(self, X, y_enc, sample_weight=None):
...
+
+        X_new = np.zeros_like(X)
+        for d in np.unique(domains):
+            idx = domains == d
+            X_new[idx] = X[idx] - self.centers_[d]
+        return X_new
+
+
+class TlTsNormalize(BaseEstimator, TransformerMixin):
+    """Normalize vectors in tangent space for transfer learning.
```
Comment (plcrodrigues) [inline pyriemann/transfer/_estimators.py:729]: Give more precision on what you mean by normalizing. Maybe say z-score ? Or explicitly say that you put mean zero and std to one.

**Review by plcrodrigues — COMMENTED** [review]
Thanks @qbarthelemy for this new feature on `pyriemann` (and sorry for the delay for the review).

The code and implementation of TSA looks good to me. Did you use the same code as the one provided by Alexandre Bleuzé? Maybe add a comment about it on the docstring?

I must admit that I would have preferred to keep the names of the classes with `TL` in caps at the beginning. And I find it a bit confusing to add `Ts` and `Ma` on the name of the class. Would it make sense (and doable) to always use the `TLCenter` class and have an option to say whether we're doing stuff on the manifold or the tangent space? (same thing for stretch and rotate). This would allow people to easily switch between operations and even create pipelines that start on manifold and finish on tangent space.

**Comment by qbarthelemy** [comment]
Thanks @plcrodrigues for your feedback. It's a good idea not to duplicate classes.

In the last version, there is a single class `TlCenter` which makes the centering in the manifold if the dimension of input is 3 (like `TLCenter`), and in the tangent space if dimension is 2 (like `TlTsCenter`). Same for `TlScale`, merging `TLStretch` and `TlTsNormalize`; and for `TlRotate`, merging `TLRotate` and `TlTsRotate`.

**Review by plcrodrigues — COMMENTED** [review]
I think it would be better to force the user to **explicitly** say whether he wants to do the centering (and stretching and rotation) on the manifold or the tangent space. Of course, the `fit` function can detect what kind of data is given depending of its dimensions, but I find the current implementation a bit too implicit and it could be annoying to debug or explain the code later. Why not instantiate the classes saying where the transformations are done? 

Also, I still prefere calling things as `TLCenter` instead of `TlCenter`, etc. Since it is an acronym, it makes more sense to me to keep things as caps lock.

**Comment by qbarthelemy** [comment]
Ok for me to keep prefix `TL`, instead of `Tl`.

**Review by sylvchev — APPROVED** [review]
Thanks @qbarthelemy for your implementation.
I'm aligned with your point of view to implicitly use the same class to handle the elements on the manifold or on the tangent space. It is the responsibility of the user to ensure that the pipelines are consistent, and we could provide some examples. I think it shows the link between the transformations, which is a good thing IMHO.

**Review by plcrodrigues — APPROVED** [review]
Looks good to me now! :-)

**Comment by plcrodrigues** [comment]
@agramfort could you squash and merge?

**Comment by sylvchev** [comment]
Thanks @qbarthelemy !
